src/tasks/WasmAppBuilder/WasmAppBuilder.cs

```diff
@@ -197,8 +197,9 @@ private bool ExecuteInternal ()
                 var tmpWebcil = Path.GetTempFileName();
                 var webcilWriter = Microsoft.WebAssembly.Build.Tasks.WebcilConverter.FromPortableExecutable(inputPath: assembly, outputPath: tmpWebcil, logger: Log);
                 webcilWriter.ConvertToWebcil();
-                var finalWebcil = Path.ChangeExtension(assembly, ".webcil");
-                FileCopyChecked(tmpWebcil, Path.Combine(asmRootPath, Path.GetFileName(finalWebcil)), "Assemblies");
+                var finalWebcil = Path.Combine(asmRootPath, Path.ChangeExtension(Path.GetFileName(assembly), ".webcil"));
+                if (Utils.CopyIfDifferent(tmpWebcil, finalWebcil, useHash: true))
+                    _fileWrites.Add(finalWebcil);
```

Comment:
```suggestion
                if (Utils.CopyIfDifferent(tmpWebcil, finalWebcil, useHash: true))
                    Log.LogMessage(MessageImportance.Log, $"Generated {finalWebcil} .");
                else
                    Log.LogMessage(MessageImportance.Log, $"Skipping generating {finalWebcil} as the contents are unchanged.");
                _fileWrites.Add(finalWebcil);
```

We want to always write to `_fileWrites` because it contributes to the incremental clean in determining what belongs as part of the build, and missing files get removed.

Though generating `_fileWrites` in the task isn't quite correct, and was my mistake. I will move that into the targets in a future PR.